api/review_checklist.md

```diff
@@ -0,0 +1,137 @@
+# API Review Checklist
+
+This checklist is intended to be used when reviewing xDS API changes.
+Users who wish to contribute API changes should read this and proactively
+consider the answers to these questions before sending a PR.
+
+## Feature Enablement
+- Are the default values going to cause behavior changes for existing users
+  who do not know about the change and have not updated the resources being
+  served by their control plane?
```

Comment:
I think much of this overlaps in spirit with the GOVERNANCE runtime guard section.  I know the API stuff is different since it affects more than just envoy, but it'd still be good to address the overlap, WDYT?

Reply:
Good point, I think we should port the relevant parts of GOVERNANCE.md across to an API GOVERNANCE.md; it might be copy+paste.

We are working towards a multi-stakeholder governance for the APIs (under CNCF, see https://github.com/cncf/udpa/blob/master/README.md), so we should evolve the API docs with that in mind.

This might seem a minor thing, but I think this decoupling is critical to ensuring that other adopters of xDS are comfortable and don't feel locked into Envoy project governance. We will do this while retaining the agree upon foundations from GOVERNANCE.md.

Reply:
Should we have a governance file in cncf/xds and just point at that from api/GOVERNANCE.md in this repo, to make it clear that it's a separate thing?  Actually, for that matter, should we do the same thing with files like this one and api/STYLE.md as well?

In any case, I agree that we should document the governance of the API and get all of these docs properly aligned, but I think a lot of this is outside the scope of this PR.  Can we follow up on that separately?

Reply:
SGTM

Reply:
SG, @markdroth can you file an issue on cncf/xds around this?

Reply:
https://github.com/cncf/xds/issues/1